src/common.rs

```diff
@@ -1123,6 +1123,35 @@ pub fn handler_invalid_tx_error(
     }
 }
 
+fn get_near_price(api: Option<url::Url>) -> color_eyre::eyre::Result<Option<f64>> {
+    let url = if let Some(api) = api {
+        api.join("api/v3/simple/price?ids=near&vs_currencies=usd")?
+    } else {
+        return Ok(None);
+    };
+
+    for _ in 0..10 {
+        let response = reqwest::blocking::get(url.clone())?;
+
+        let body = response.text()?;
+        let parsed_body: serde_json::Value = serde_json::from_str(&body)?;
```

Comment:
Instead of getting `.text()` and then manually parsing it, use `.json()` method.

```rust
if let Ok(parsed_body) = response.json::<serde_json::Value>()
```